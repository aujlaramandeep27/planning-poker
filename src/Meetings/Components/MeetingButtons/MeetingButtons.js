import React from 'react';
import { connect } from 'react-redux';
import { viewMeeting } from '../../../Navigation/route-actions';

export function MeetingButtons({ meetings = [], goToMeeting }) {
  return meetings.map((meeting, index) =>
    <div className="uk-margin-small" key={ index }>
      <div className="uk-card uk-card-secondary uk-card-body pp-button" onClick={ () => goToMeeting(meeting._id) }>
        <h3 className="uk-card-title">{ meeting.name }</h3>
      </div>
    </div>
  );
}

function mapStateToProps(state) {
  return {
    meetings: state.meetings,
    error: state.error
  }
}

function mapDispatchToProps(dispatch) {
  return {
    goToMeeting: (meetingId) => dispatch(viewMeeting(meetingId))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(MeetingButtons)
