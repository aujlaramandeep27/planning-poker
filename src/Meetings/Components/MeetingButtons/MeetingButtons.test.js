import {MeetingButtons} from "./MeetingButtons";

it('MeetingButtons renders correctly', () => {
    const props = {
        meetings: [{_id: 'id', name: 'test'}],
        error: null,
        goToMeeting: jest.fn()
    };

    expect(MeetingButtons(props)).toMatchSnapshot();
});
