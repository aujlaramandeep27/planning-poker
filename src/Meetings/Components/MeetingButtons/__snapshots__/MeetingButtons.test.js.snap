// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetingButtons renders correctly 1`] = `
Array [
  <div
    className="uk-margin-small"
  >
    <div
      className="uk-card uk-card-secondary uk-card-body pp-button"
      onClick={[Function]}
    >
      <h3
        className="uk-card-title"
      >
        test
      </h3>
    </div>
  </div>,
]
`;
